### PR TITLE
Only show Lesson Extras field for CSF courses

### PR DESCRIPTION
**Pull request by joshlory**

### CODE REVIEW

File: apps/src/templates/teacherDashboard/teacherSectionsRedux.js [hunk 1]
```diff
@@ -398,6 +400,13 @@ export default function teacherSections(state=initialState, action) {
     };
   }
 
+  if (action.type === SET_CSF_SCRIPT_IDS) {
+    return {
+      ...state,
+      isCsfScript: id => action.ids.indexOf(id) > -1,
```
Comment (islemaster) [inline apps/src/templates/teacherDashboard/teacherSectionsRedux.js]: Code smell: Functions in Redux state.

[From the Redux FAQ](https://redux.js.org/faq/organizing-state#can-i-put-functions-promises-or-other-non-serializable-items-in-my-store-state):

> **Can I put functions, promises, or other non-serializable items in my store state?**
> It is highly recommended that you only put plain serializable objects, arrays, and primitives into your store. It's technically possible to insert non-serializable items into the store, but doing so can break the ability to persist and rehydrate the contents of a store, as well as interfere with time-travel debugging.
>
> If you are okay with things like persistence and time-travel debugging potentially not working as intended, then you are totally welcome to put non-serializable items into your Redux store.  Ultimately, it's your application, and how you implement it is up to you.  As with many other things about Redux, just be sure you understand what tradeoffs are involved.

Obviously we're not using persistence or time-travel debugging in our application, but I worry that patterns like this are likely to interfere with upgrades down the road - either newer versions of Redux that enforce stricter rules for the sake of performance optimizations as we've seen a few times with React, or with future attempts by our team to introduce some related tool (like prehydrating a redux store on the server, for instance) that would require stricter guidelines.

Instead, I would export a specialized selector function and call it in your `mapStateToProps` function when connecting `EditSectionForm`.

```js
// teacherSectionsRedux.js
export const isCsfScript = (state, id) => state.csfScriptIds.indexOf(id) > -1;

// EditSectionForm.jsx
export default connect(state => ({
  isCsfScript: id => isCsfScript(state, id)
}))
```

Comment (joshlory) [reply apps/src/templates/teacherDashboard/teacherSectionsRedux.js]: Good catch, updated in a1ab5d2fdf924d3a834ff862f9a988633dd407cf.

**Comment by islemaster** [comment]
@davidsbailey who might have thoughts on my comment above, in the context of curriculum versioning.

**Comment by joshlory** [comment]
I'm not sure there's much of a practical difference in putting the data in the bundle, since we cache the list and inject it into the page that needs the data (no extra API call).  But happy to structure this either way.

**Comment by joshlory** [comment]
One benefit would be ease of including the helper on other pages (don't need to first call `setCsfScriptIds` to put the data in the store).  Now I'm not sure which approach I like better 😛.

**Comment by joshlory** [comment]
Moving ahead with the Redux pattern for now.

**Pull request merged**